### ISSUE TITLE
Is that necessary use 'db.session.add(self)' in function get_token?

### ISSUE THREAD

**Issue by Exmlyshy** [miguelgrinberg/microblog#154]
Hi,miguel.

In `app.models.User.get_token()` [models](https://github.com/miguelgrinberg/microblog/blob/master/app/models.py)，I feel puzzled that you use `db.session.add(self)` after set a token , but you just use `db.session.commit()` in another function relate to modify a user's info and use `db.session.add` when create a new user.So when to use or not use `add`?thanks.

**Comment by miguelgrinberg** [comment]
@Exmlyshy The `db.session.add()` includes the object in the session. In most cases, the object that you are using is already in the session because you used a query to obtain it, so that is why in many places I do not call `add()` before a commit.

In general my choice of putting an explicit `add()` has to do with readability more than it being required. In this case with the token it is unlikely that removing the `add()` will cause a problem, but given that when you look at the function you do not see a previous query that obtains the user, in my view it makes sense to be explicit so that when you read the code you have no doubt that the object will be committed.

**Comment by Exmlyshy** [comment]
Thanks. 

> In most cases, the object that you are using is already in the session because you used a query to obtain it

I don't know how to get an existing user without using query , I thought may be `user.followers`?so I did a test. There is a user A, A has follower B.
```
>>> A=User.query.get(1)
>>> B=A.followers.first()
>>> B.token
>>> B.token=1
>>> db.session.commit()
```
without `db.session.add(B)` , I found it works, in another terminal `B.token==1`
when I got a User instance A from a query, it's in the session and instances from A.relationships also in the session .Maybe I should go to learn some about sqlalchemy , thank you.
